models/intermediate/int_klaviyo__event_attribution.sql

```diff
@@ -100,18 +100,70 @@ attribute as (
     from last_touches 
 ),
 
-final as (
-
+order_seq as (
     select
-        *,
-
-        -- get whether the event is attributed to a flow or campaign
-        coalesce(touch_type, first_value(touch_type) over(
-            partition by person_id, source_relation, touch_session order by occurred_at asc rows between unbounded preceding and current row)) 
+        attribute.*,
+        -- stable ordering in case of same-time events
+        sum(case when lower(attribute.type) = 'placed order' then 1 else 0 end) over (
+            partition by attribute.person_id, attribute.source_relation
+            order by attribute.occurred_at, attribute.unique_event_id
+            rows between unbounded preceding and current row
+        ) as placed_order_group
+    from attribute
+),
 
-            as session_touch_type -- if the session events qualified for attribution, extract the type of touch they are attributed to
+placed_index as (
+    -- one row per Placed Order carrying the touch to copy
+    select
+        order_seq.person_id,
+        order_seq.source_relation,
+        order_seq.placed_order_group,
+        order_seq.last_touch_id as placed_order_touch_id
+    from order_seq
+    where lower(order_seq.type) = 'placed order'
+    ),
+
+adjust_orders as (
+    select
+        order_seq.*,
+        case
+            when 
+                lower(order_seq.type) in ('fulfilled order', 'fulfilled partial order', 'delivered shipment', 'marked out for delivery', 'confirmed shipment', 'refunded order') 
+                and order_seq.placed_order_group > 0
+            then placed_index.placed_order_touch_id
+            else order_seq.last_touch_id
+        end as adjusted_last_touch_id
+    from order_seq
+    left join placed_index
```

Comment:
I wonder if we should place some sort of time limit, like the `'placed order'` event needs to have happened within the previous X days/weeks/months. Just in case `'fulfilled order'` or any of these other event types were logged without an order placement for some reason (not sure how possible/common this is though). 

@fivetran-catfritz thoughts?

Reply:
That makes sense. I added line 135 and set it at 3 months for now. I tested it again and no change as expected in the case I was looking at. What do you think?